app/app.rb

```diff
@@ -0,0 +1,118 @@
+ENV['ENVIRONMENT'] ||= 'development'
+
+require 'sinatra/base'
+require 'sinatra/flash'
+require './app/models/message'
+require './app/models/user'
+require './app/models/tag'
+require './app/models/mail'
+require './db_setup'
+require './lib/seeds'
+require 'mail'
+
+
+class Chitter < Sinatra::Base
+  enable :sessions, :method_override
+  set :session_secret, 'super secret'
+  register Sinatra::Flash
+
+  get '/' do
+    erb :login, :layout => :layout
+  end
+
+  get '/signup' do
+    erb :signup, :layout => :layout
+  end
+
+  post '/signup' do
+    if params[:password] != params[:confirm_password]
+      flash[:notice] = 'The passwords do not match. Please try again.'
+      redirect '/signup'
+    end
+    
+    user = User.create(username:params[:username], password:params[:password], name:params[:name], email:params[:email])
+    
+    if user
+      flash[:notice] = 'The account has been successfully created. Please log in.'
+      redirect '/'
+    else 
+      flash[:notice] = 'The email or username is already taken. Please try another one.'
+      redirect '/signup'
+    end
+  end
+
+  post '/sessions' do
+    user = User.authenticate(params[:email], params[:password])
+    if user
+      session[:user] = user.email
+      flash[:notice] = "Welcome back, #{user.username}!"
+      redirect '/home'
+    else
+      flash[:notice] = "The email or password did not match our records. Please try again."
+      redirect '/'
+    end
+  end
+
+  get '/follow/:username' do
+    flash[:notice] = "You are now following @#{params[:username]}"
```

Comment:
I like the use of all the flash notices to the user to let them know what theyve done and how the process went